File: parse_pdfs.py
import PyPDF2
import os

docs_dir = r"C:\Users\Enrique Saavedra\Documents\Programa ACT\Documentos"
files = [
  'ACT-45 Actividades.pdf',
  'ACT-45 Instrucciones.pdf',
  'ACT-96 Inspeccion.pdf',
  'ACT-96 Instrucciones.pdf'
]

for file in files:
    try:
        path = os.path.join(docs_dir, file)
        out_path = os.path.join(docs_dir, file.replace('.pdf', '.txt'))
        with open(path, 'rb') as f:
            reader = PyPDF2.PdfReader(f)
            text = ""
            for page in reader.pages:
                text += page.extract_text() + "\n"
        with open(out_path, 'w', encoding='utf-8') as f:
            f.write(text)
        print(f"Success {file}")
    except Exception as e:
        print(f"Error {file}: {e}")

File: parse_pdfs.js
const fs = require('fs');

const path = require('path');

const docsDir = path.join(__dirname, 'Documentos');
const files = [
  'ACT-45 Actividades.pdf',
  'ACT-45 Instrucciones.pdf',
  'ACT-96 Inspeccion.pdf',
  'ACT-96 Instrucciones.pdf'
];

const PDFParser = require("pdf2json");

async function parsePdfs() {
  for (const file of files) {
    try {
      const filePath = path.join(docsDir, file);
      if (!fs.existsSync(filePath)) {
        console.error(`File not found: ${filePath}`);
        continue;
      }
      
      const pdfParser = new PDFParser(null, 1);
      
      const text = await new Promise((resolve, reject) => {
        pdfParser.on("pdfParser_dataError", errData => reject(errData.parserError));
        pdfParser.on("pdfParser_dataReady", pdfData => {
          resolve(pdfParser.getRawTextContent());
        });
        pdfParser.loadPDF(filePath);
      });

      const outPath = path.join(docsDir, file.replace('.pdf', '.txt'));
      fs.writeFileSync(outPath, text);
      console.log(`Successfully parsed ${file} to ${outPath}`);
    } catch (err) {
      console.error(`Error parsing ${file}:`, err);
    }
  }
}


parsePdfs();
